starter_code/queries.md

```diff
@@ -4,76 +4,157 @@
 
-<!-- Your Code Goes Here -->
+```js
+db.companies.find().sort({"ipo.valuation_amount": -1}).limit(20) 
+//needed to limit due to memory error
+```
+
+
 
 ### 10. Retrieve the 10 companies with more employees, order by the `number of employees`
 
-<!-- Your Code Goes Here -->
+```js
+db.companies.find().sort({number_of_employees: -1}).limit(10)
+
+```
+
+
 
 ### 11. All the companies founded on the second semester of the year. Limit your search to 1000 companies.
 
-<!-- Your Code Goes Here -->
+```js
+db.companies.find({founded_month: {$gt: 6}}).limit(1000)
+
+```
+
+
 
 ### 12. All the companies founded before 2000 that have an acquisition amount of more than 10.000.000
 
-<!-- Your Code Goes Here -->
+```js
+db.companies.find({$and: [{founded_year:{$lt: 2000}}, {"acquisition.price_amount": {$gt:10000000}}] })
+```
+
+
 
 ### 13. All the companies that have been acquired after 2010, order by the acquisition amount, and retrieve only their `name` and `acquisition` field.
 
-<!-- Your Code Goes Here -->
+```js
+db.companies.find({"acquisition.acquired_year":{$gt:2010}},{name:1, acquisition:1,_id:0}).sort({acquisition:1})
+```
+
+
 
 ### 14. Order the companies by their `founded year`, retrieving only their `name` and `founded year`.
 
-<!-- Your Code Goes Here -->
+```js
+db.companies.find({},{name:1, founded_year:1, _id:0}).sort({founded_year:1}).limit(20)
+```
+
+
 
 ### 15. All the companies that have been founded on the first seven days of the month, including the seventh. Sort them by their `acquisition price` in a descending order. Limit the search to 10 documents.
 
-<!-- Your Code Goes Here -->
+```js
+db.companies.find().sort({"acquisition.acquired_day":{$lte: 7}}).sort({"acquisition.price_amount":-1}).limit(10)
+```
+
+
 
 ### 16. All the companies on the 'web' `category` that have more than 4000 employees. Sort them by the amount of employees in ascending order.
 
-<!-- Your Code Goes Here -->
+```js
+db.companies.find({$and: [{category_code: "web"}, {number_of_employees: {$gt:4000}}]}).sort({number_of_employees: 1})
+```
+
+
 
 ### 17. All the companies whose acquisition amount is more than 10.000.000, and currency is 'EUR'.
 
-<!-- Your Code Goes Here -->
+```js
+db.companies.find({$and:[{"acquisition.price_amount": {$gt:10000000}}, {"acquisition.price_currency_code": "EUR"}]})
+```
+
+
 
 ### 18. All the companies that have been acquired on the first trimester of the year. Limit the search to 10 companies, and retrieve only their `name` and `acquisition` fields.
 
-<!-- Your Code Goes Here -->
+```js
+db.companies.find({"acquisition.acquired_month": {$lte: 3}},{name:1,acquisition:1}).limit(10)
+```
+
+
 
 ### 19. All the companies that have been founded between 2000 and 2010, but have not been acquired before 2011.
 
-<!-- Your Code Goes Here -->
+```js
+db.companies.find({$and:[{founded_year: {$gt:2000}},{founded_year: {$lt:2010}}, {"acquisition.acquired_year": {$gt:2011}}]})
+```
```

Comment:
aqui puedes incluir todos los años con los operadores $gte y $lte
`db.companies.find({$and: [{founded_year: {$gte: 2000}},{founded_year: {$lte: 2010}},{"acquisition.acquired_year": {$gte: 2011}}]}, {name: 1, "acquisition.acquired_year": 1})`